crates/cast/bin/opts.rs

```diff
@@ -330,6 +330,8 @@ pub enum CastSubcommand {
     /// Get the latest block number.
     #[command(visible_alias = "bn")]
     BlockNumber {
+        /// The hash or tag to query. If not specified, the latest number is returned.
+        block: Option<BlockId>,
         #[command(flatten)]
         rpc: RpcOpts,
     },
```

Comment:
could we add a few parse tests for this,

see for reference:

https://github.com/foundry-rs/foundry/blob/master/crates/cast/bin/cmd/wallet/vanity.rs#L364-L371